api/http/templates.go

```diff
@@ -74,229 +74,1311 @@ const bzzList = `{{ define "content" }}
+						<span class="controlHeaderItemIcon">
+							<svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
+								 viewBox="0 0 20 20" style="enable-background:new 0 0 20 20;" xml:space="preserve">
+							<path d="M10,0C4.5,0,0,4.5,0,10c0,5.5,4.5,10,10,10s10-4.5,10-10C20,4.5,15.5,0,10,0z M10,4.4c0.9,0,1.7,0.8,1.7,1.7
+								S10.9,7.8,10,7.8S8.3,7.1,8.3,6.1S9.1,4.4,10,4.4z M12.3,14.7c0,0.3-0.2,0.5-0.5,0.5H8.2c-0.3,0-0.5-0.2-0.5-0.5v-1
+								c0-0.3,0.2-0.5,0.5-0.5h0.5v-2.6H8.2c-0.3,0-0.5-0.2-0.5-0.5v-1c0-0.3,0.2-0.5,0.5-0.5h2.6c0.3,0,0.5,0.2,0.5,0.5v4h0.5
+								c0.3,0,0.5,0.2,0.5,0.5V14.7z"/>
+							</svg>
+						</span>
+					</a>
+				</div>
+			</div>
+			<div class="controlMain">
+				<div class="controlComponent hidden fades" id="downloadComponent">
+					<div class="controlComponentMessage">
+						Enter a hash or ENS name.
+					</div>
+					<form id="downloadForm" spellcheck="false">
+						<input type="text" id="downloadHashField" name="downloadHash" placeholder="theswarm.eth"/>
+						<button type="submit" value="Go" class="orangeButton wideButton">
+							Go
+							<svg version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
+								 viewBox="0 0 20 20" style="enable-background:new 0 0 16 20;" xml:space="preserve">
+								<path id="XMLID_1617_" d="M10,0C4.5,0,0,4.5,0,10s4.5,10,10,10c5.5,0,10-4.5,10-10C20,4.5,15.5,0,10,0z M9.8,17.9
+			c-4.4,0-7.9-3.6-7.9-7.9c0-4.4,3.6-7.9,7.9-7.9c0.4,0,0.9,0,1.3,0.1l0,0.8c0,0.3-0.3,0.6-0.6,0.6l-1,0C9.3,3.6,9.1,3.7,9,3.9
+			L8.7,4.4C8.6,4.5,8.5,4.6,8.3,4.6L7.7,4.8C7.5,4.8,7.3,5.1,7.3,5.4l0,0.2c0,0.3,0.3,0.6,0.6,0.6l3.6,0c0.2,0,0.3,0.1,0.4,0.2
+			l0.3,0.3c0.1,0.1,0.3,0.2,0.4,0.2l0.4,0c0.3,0,0.6,0.3,0.6,0.6c0,0.3-0.2,0.5-0.4,0.6l-1.9,0.6c-0.2,0.1-0.3,0-0.5,0l-0.6-0.3
+			c-0.3-0.2-0.6-0.2-1-0.2h0C8.8,8.1,8.3,8.2,8,8.5L6.9,9.3C6.3,9.8,6,10.4,6,11.1l0,0.6c0,0.6,0.2,1.1,0.6,1.5
+			c0.4,0.4,1,0.6,1.5,0.6l1,0c0.3,0,0.6,0.3,0.6,0.6l0,1.2c0,0.5,0.1,1,0.3,1.4c0.2,0.4,0.6,0.6,1,0.6c0.4,0,0.7-0.2,0.9-0.5l0.5-0.8
+			c0.3-0.4,0.6-0.8,1-1.2c0.1-0.1,0.2-0.2,0.2-0.3L14,14c0-0.1,0.1-0.2,0.1-0.3l0.7-1c0.1-0.1,0.1-0.2,0.1-0.4l0-0.5
+			c0-0.3-0.3-0.6-0.6-0.6l-0.3,0c-0.2,0-0.4-0.1-0.5-0.3L13,10.2c-0.2-0.3-0.1-0.8,0.3-0.9l0.1,0c0.2-0.1,0.4,0,0.5,0.1l0.7,0.5
+			c0.2,0.1,0.4,0.1,0.6,0l0.6-0.3c0.2-0.1,0.3-0.3,0.3-0.6l0-0.3c0-0.3,0.3-0.6,0.6-0.6l0.7,0c0.2,0.6,0.2,1.3,0.2,1.9
+			C17.8,14.4,14.2,17.9,9.8,17.9z"/>
+							</svg>
+						</button>
+					</form>				
+				</div>
+				<div class="controlComponent fades" id="uploadComponent">
+					<div class="controlComponentMessage">
+						Select a file to upload it to the Swarm network.
+					</div>				
+					<form id="uploadForm" enctype="multipart/form-data" spellcheck="false">
+						<input type="text" id="uploadSelectedFile" name="uploadSelected" placeholder="select file or directory..."/>
```

Comment:
I am not able to select a directory in Chrome 77 on macOS. Also not sure if directory upload would work.

Reply:
i have not included directory uploads in this release, but agree it is useful functionality and that we should explore in a feature request and implement soon

Reply:
agree - let's defer this to a later PR - https://github.com/ethersphere/swarm/issues/1868